### PR TITLE
refactor(scraper): Uses a faster title-dedupe scraping routine for XML

### PR DESCRIPTION
**Pull request by BossRighteous**
Here's the PR description:

---

### Summary

The gamelist.xml scraper previously iterated indexed `Media` rows and matched each against gamelist entries via path normalization (exact path + filename fallback). For large collections this was O(media × gamelist_entries) per system per root, with two hash maps built per gamelist file.

This PR inverts the loop: `scrapeLoop` now pre-builds a `slug → MediaTitle` map (unscraped titles only, or all titles in Force mode) using `FindMediaTitlesWithoutSentinel` before calling `LoadRecords`. `LoadRecords` then iterates gamelist entries and does an O(1) slug lookup per entry, deleting matched slugs to enforce first-match-wins deduplication across multiple ROM roots.

Key structural changes:
- `LoadRecords` signature gains `titlesBySlug` and `mediaByTitleDBID` params; DB queries moved to caller (`scrapeLoop`)
- Pre-filtering via `FindMediaTitlesWithoutSentinel` replaces the post-load `scrapedMediaIDs` set — unscraped titles arrive as the lookup key set rather than being filtered after matching
- `buildGameIndexes`, `normalizeMediaPath`, and `gamelistEntry` removed
- `GamelistRecord.MatchedPath` removed — `MapToDB` uses DB IDs only
- `cfg *config.Instance` added to `GamelistXMLScraper` for slug computation via `mediascanner.GetPathFragments`
- `MapToDB` drops media-level lang/region tags; this scraper now writes title-level metadata only

### Tests

- `TestLoadRecords_SlugMatch` — slug match produces correct DB IDs
- `TestLoadRecords_FirstWins` — duplicate slug across roots: first root wins
- `TestLoadRecords_NoMediaForTitle` — slug match with no Media row yields `MatchedMediaDBID = 0`
- `TestLoadRecords_ContextCancellation` — cancelled context returns `context.Canceled`
- `TestLoadRecords_PathTraversalSkipped` — `../../etc/passwd` entries skipped; valid entry still matches
- `TestScrapingStatus_ClearIfOwner_*` — owner/non-owner clear semantics
- `TestHandleMediaScrape_EmitsProgressUpdates` — progress notifications reflect actual processed count
- `TestKeyboardControls_*` — `KeyboardControls` builder and error propagation

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved gamelist.xml scraping to use slug-based matching for more reliable game detection.
  * Enhanced handling of malformed or missing gamelist files during scraping.

* **Tests**
  * Added comprehensive test coverage for media scraper resume behavior.
  * Expanded test suite for gamelist.xml scraping including slug-matching validation and edge case handling.
  * Added keyboard controls functionality tests.

### CODE REVIEW

File: pkg/database/scraper/gamelistxml/scraper.go [hunk 1]
```diff
@@ -522,18 +510,15 @@ func (g *GamelistXMLScraper) scrapeLoop(
 }
 
 // MapToDB converts a GamelistRecord into the tag and property writes to apply
-// to the matched Media and MediaTitle rows.
+// to the matched MediaTitle row. Media-level tags are not written by this
+// scraper; only title-level tags and properties are populated.
 func (g *GamelistXMLScraper) MapToDB(record *GamelistRecord) scraper.MapResult {
```
[the remark this answers is not in the document]
Comment (BossRighteous) [reply pkg/database/scraper/gamelistxml/scraper.go:523]: Not a concern in this phase. I want there to be additive behavior until a full scrape option is implemented.
